run.py

```diff
@@ -100,50 +118,51 @@ def nonbib_delta_to_master_pipeline(schema, batch_size=1):
         task_output_results.delay(recs)
 
 
-def metrics_to_master_pipeline(schema, batch_size=1):
+def metrics_to_master_pipeline(metrics_engine, schema, batch_size=1):
     """send all metrics data to queue for delivery to master pipeline"""
-    m = metrics.Metrics(schema)
-    connection = m.engine.connect()
-    s = select([m.table])
-    results = connection.execute(s)
+    global config
+    Session = sessionmaker(bind=metrics_engine)
+    session = Session()
+    session.execute('set search_path to {}'.format(schema))
     tmp = []
-    for current_row in results:
-        current_row = dict(current_row)
+    i = 0
+    max_rows = config['MAX_ROWS']
+    for current_row in session.query(models.MetricsTable).yield_per(100):
+        current_row = row2dict(current_row)
         current_row.pop('id')
-        logger.debug('Will forward this metrics record: %s', current_row)
         rec = MetricsRecord(**current_row)
         tmp.append(rec._data)
+        i += 1
+        if max_rows > 0 and i > max_rows:
+            break
         if len(tmp) >= batch_size:
             recs = MetricsRecordList()
             recs.metrics_records.extend(tmp)
-            logger.debug("Calling metrics 'app.forward_message' with '%s' records", len(recs.metrics_records))
+            logger.info("Calling metrics 'app.forward_message' count = '%s'", i)
             task_output_metrics.delay(recs)
             tmp = []
 
     if len(tmp) > 0:
         recs = MetricsRecordList()
         recs.metrics_records.extend(tmp)
-        logger.debug("Calling metrics 'app.forward_message' with final '%s' records", len(recs.metrics_records))
+        logger.debug("Calling metrics 'app.forward_message' with count = '%s'", i)
         task_output_metrics.delay(recs)
 
 
-
-def metrics_delta_to_master_pipeline(metrics_schema, nonbib_schema, batch_size=1):
+def metrics_delta_to_master_pipeline(metrics_engine, metrics_schema, nonbib_schema, batch_size=1):
     """send data for changed metrics to master pipeline
 
     the delta table was computed by comparing to sets of nonbib data
     perhaps ingested on succesive days"""
     m = metrics.Metrics(metrics_schema)
-    nonbib = row_view.SqlSync(nonbib_schema)
+    nonbib = nonbib.NonBib(nonbib_schema)
```

Comment:
so this is confusing; it overrides module (in the function scope) with the object

Reply:
addressed in new commit: https://github.com/spacemansteve/AdsDataSqlSync/commit/04a256b429c48c444a8984d69487f50ee6aff8b2